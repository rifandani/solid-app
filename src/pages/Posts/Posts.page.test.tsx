import { Route } from '@solidjs/router';
import { waitFor } from '@solidjs/testing-library';
import { mockedCreateResource } from '../../mocks/module.mock';
import { renderWithRouter } from '../../utils/test.util';
import PostsPage from './Posts.page';

mockedCreateResource.mockReturnValueOnce([
  () => ({
    title: 'sunt aut',
    body: 'quia et',
    user: {
      name: 'Leanne Graham',
    },
  }),
]);

describe('PostsPage', () => {
  it('should render correctly', async () => {
    // ARRANGE
    const { container, getByText } = renderWithRouter(() => (
      <Route path="/" component={PostsPage} /> // the path doesn't matter here
    ));

    // ASSERT
    await waitFor(() => {
      expect(mockedCreateResource).toHaveBeenCalled();
      expect(getByText(/Post List/)).toBeInTheDocument();
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
