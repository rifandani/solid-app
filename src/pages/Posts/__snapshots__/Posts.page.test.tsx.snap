// Vitest Snapshot v1

exports[`PostsPage > should render correctly 1`] = `
<main
  class="py-20 px-10 md:px-24 lg:px-40 xl:px-52 flex flex-col items-center justify-center"
>
  <h1
    class="text-2xl font-semibold tracking-wider text-violet-500"
  >
    Post List
  </h1>
  <button
    class="my-5"
  >
    Add Post
  </button>
  <section
    class="w-full gap-4 grid grid-cols-1 sm:grid-cols-2 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4"
  >
    
    
  </section>
</main>
`;
